AWSS3/AWSS3TransferUtility.m

```diff
@@ -2433,7 +2432,12 @@ - (void)URLSession:(NSURLSession *)session
     
         //Calculate the total sent so far
         int64_t totalSentSoFar = 0;
-        for (AWSS3TransferUtilityUploadSubTask *aSubTask in [transferUtilityMultiPartUploadTask.completedPartsDictionary allValues]) {
+        //Create a new Dictionary with the partNumber as the Key
+        NSMutableDictionary *completedPartsByPartNumber = [NSMutableDictionary new];
+        for(AWSS3TransferUtilityUploadSubTask *subTask in transferUtilityMultiPartUploadTask.completedPartsSet) {
+            [completedPartsByPartNumber setObject:subTask forKey:subTask.partNumber];
+        }
+        for (AWSS3TransferUtilityUploadSubTask *aSubTask in [completedPartsByPartNumber allValues]) {
```

Comment:
i'm being defensive here in case the `completedPartsSet` contains duplicate partNumber, which is populated when the app calls `recovery` and hydrate the data from DB. I don't know exactly if there would be, but best be defensive about this, considering we are fixing a similar bug with this change. Another consideration was to turn the completedParts dictionary to key off of the partNumber. But with this solution with the Set, we should make sure to handle any items with the same partNumber